.docker/bin/hubot

```diff
@@ -0,0 +1,177 @@
+██████╔╝██║   ██║██║     █████╔╝ █████╗     ██║      ██║     ███████║███████║   ██║   
+██╔══██╗██║   ██║██║     ██╔═██╗ ██╔══╝     ██║      ██║     ██╔══██║██╔══██║   ██║   
+██║  ██║╚██████╔╝╚██████╗██║  ██╗███████╗   ██║   ██╗╚██████╗██║  ██║██║  ██║   ██║   
+╚═╝  ╚═╝ ╚═════╝  ╚═════╝╚═╝  ╚═╝╚══════╝   ╚═╝   ╚═╝ ╚═════╝╚═╝  ╚═╝╚═╝  ╚═╝   ╚═╝   
+                ██╗  ██╗██╗   ██╗██████╗  ██████╗ ████████╗
+                ██║  ██║██║   ██║██╔══██╗██╔═══██╗╚══██╔══╝
+                ███████║██║   ██║██████╔╝██║   ██║   ██║   
+                ██╔══██║██║   ██║██╔══██╗██║   ██║   ██║   
+                ██║  ██║╚██████╔╝██████╔╝╚██████╔╝   ██║   
+                ╚═╝  ╚═╝ ╚═════╝ ╚═════╝  ╚═════╝    ╚═╝   
+EOF
+
+echo "Your Rocket.Chat Hubot Docker Container is now starting. Please wait...."
+
+#Set log level to info by default if it has not been set
+if [[ -z "${HUBOT_LOG_LEVEL}" ]]; then
+	echo "INFO: HUBOT_LOG_LEVEL not set. Using HUBOT_LOG_LEVEL=info as the default"
+	export HUBOT_LOG_LEVEL=info
+fi
+#Check if the Rocket.Chat URL has been set.
+if [[ -z "${ROCKETCHAT_URL}" ]]; then
+	echo "-------------"
+	echo "The ROCKETCHAT_URL Environment Variable has not been set. Set this to your Rocket.Chat Server URL"
+	echo "Example: ROCKETCHAT_URL=https://open.rocket.chat"
+	echo "Exiting...."
+	echo "-------------"
+	exit 1
+fi
+
+#Check if the Rocket.Chat User has been set.
+if [[ -z "${ROCKETCHAT_USER}" ]]; then
+	echo "-------------"
+	echo "ROCKETCHAT_USER Environment Variable has not been set. Set this to the username on your Rocket.Chat server that the bot will log in with."
+	echo "Example: ROCKETCHAT_USER=rocketbot"
+	echo "Exiting...."
+	echo "-------------"
+	exit 1
+fi
+
+#Check if the Rocket.Chat Password has been set for the user above
+if [[ -z "${ROCKETCHAT_PASSWORD}" ]]; then
+	echo "-------------"
+        echo "----ERROR----"
+        echo "-------------"
+	echo "The ROCKETCHAT_PASSWORD Environment Variable has not been set. Set this to the password for the user specified in the ROCKETCHAT_USER environment variable."
+	echo "Example: ROCKETCHAT_PASSWORD=supersecret"
+	echo "Exiting...."
+	echo "-------------"
+	exit 1
+
+fi
+
+#Check for BOT_NAME or HUBOT_NAME.
+if [[ -z "${BOT_NAME}" ]]; then
+	if [[ -z "${HUBOT_NAME}" ]]; then
+		echo "-------------"
+		echo "ERROR: The BOT_NAME Environment Variable has not been set. Set this to the name your bot will respond too."
+		echo "Exiting...."
+		echo "-------------"
+		exit 1
+	fi
+else
+	export HUBOT_NAME=${BOT_NAME}
+fi
+
+#Install any required deps.
+cd /home/hubot/
+
+if [[ -z "${NPM_REGISTRY}" ]]; then
+        echo "INFO: The NPM_REGISTRY environment variable has not been set. Using npmjs as the default."
+else 
+        echo "INFO: The NPM_REGISTRY environment variable is $NPM_REGISTRY. NPM will use this registry to pull packages from."
+        npm set registry $NPM_REGISTRY
+fi
+
+echo "INFO: Attempting to install this containers dependancies"
+npm install --no-audit
```

Comment:
Do we need this since you do during image build?

Reply:
2 reasons 

This is to combat a really strange issue I was seeing where during build it was not installing 1 or 2 random dependencies . Tried with yarn and npm, no joy. The extra 5 seconds this takes to just make sure is worth it in my eyes 

And 

When people specify new scripts with the EXTERNAL_SCRIPTS env var those deps are not installed at build and will need to be installed. As most won't be building it merely deploying from docker hub

Reply:
Actually, the EXTERNAL_SCRIPTS is done separately 

Ignore that second reason

Mainly the first, happy to try again without it but was seeing really strange dep issues and added it in as a just in case